Build failure: linuxKernel.packages.linux_6_14.apfs

### Nixpkgs version

- Unstable (25.05)

### Steps to reproduce

Set your kernel to the latest release with `boot.kernelPackages = pkgs.linuxPackages_latest;` (6.14 at time of writing), include the `apfs` kernel module with `kernelModules` and `extraModulePackages`, and rebuild the system, apfs module will fail to build due to breaking upstream kernel change.

### Can Hydra reproduce this build failure?

No, Hydra cannot reproduce this build failure.

### Link to Hydra build job

_No response_

### Relevant log output

```console
error: builder for '/nix/store/b14dxf795q2i7mjq8v06ikk4v4ramyj7-apfs-0.3.12-6.14.drv' failed with exit code 2;
       last 25 log lines:
       >   CC [M]  btree.o
       >   CC [M]  compress.o
       >   CC [M]  dir.o
       >   CC [M]  extents.o
       >   CC [M]  file.o
       >   CC [M]  inode.o
       >   CC [M]  key.o
       >   CC [M]  libzbitmap.o
       >   CC [M]  lzfse/lzfse_decode.o
       >   CC [M]  lzfse/lzfse_decode_base.o
       >   CC [M]  lzfse/lzfse_fse.o
       >   CC [M]  lzfse/lzvn_decode_base.o
       >   CC [M]  message.o
       >   CC [M]  namei.o
       > namei.c:139:27: error: initialization of 'int (*)(struct inode *, const struct qstr *, struct dentry *, unsigned int)' from incompatible pointer type 'int (*)(struct dentry *, unsigned int)' [-Wincompatible-pointer-types]
       >   139 |         .d_revalidate   = apfs_dentry_revalidate,
       >       |                           ^~~~~~~~~~~~~~~~~~~~~~
       > namei.c:139:27: note: (near initialization for 'apfs_dentry_operations.d_revalidate')
       > make[4]: *** [/nix/store/1v70gd12v5chh69vqb0ld3xaqaz298sn-linux-6.14-dev/lib/modules/6.14.0/source/scripts/Makefile.build:207: namei.o] Error 1
       > make[3]: *** [/nix/store/1v70gd12v5chh69vqb0ld3xaqaz298sn-linux-6.14-dev/lib/modules/6.14.0/source/Makefile:1994: .] Error 2
       > make[2]: *** [/nix/store/1v70gd12v5chh69vqb0ld3xaqaz298sn-linux-6.14-dev/lib/modules/6.14.0/source/Makefile:251: __sub-make] Error 2
       > make[2]: Leaving directory '/build/source'
       > make[1]: *** [/nix/store/1v70gd12v5chh69vqb0ld3xaqaz298sn-linux-6.14-dev/lib/modules/6.14.0/source/Makefile:251: __sub-make] Error 2
       > make[1]: Leaving directory '/nix/store/1v70gd12v5chh69vqb0ld3xaqaz298sn-linux-6.14-dev/lib/modules/6.14.0/build'
       > make: *** [Makefile:18: default] Error 2
       For full logs, run 'nix log /nix/store/b14dxf795q2i7mjq8v06ikk4v4ramyj7-apfs-0.3.12-6.14.drv'.
error: 1 dependencies of derivation '/nix/store/5r34j1lbmw32zgnah9rdmvxdvajpaipd-linux-6.14-modules.drv' failed to build
error: 1 dependencies of derivation '/nix/store/582r7advc0fb18158ga0fqlri52j3s9b-nixos-system-ryuu-25.05.20250327.5e5402e.drv' failed to build
```

### Additional context

Issue was resolved upstream, apfs package just needs to be bumped up to latest commit or the v0.3.13 tag. See linux-apfs/linux-apfs-rw#88, linux-apfs/linux-apfs-rw@e16e7f4, linux-apfs/linux-apfs-rw@61d8e5b

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.1, NixOS, 25.05 (Warbler), 25.05.20241203.55d15ad`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.10`
 - nixpkgs: `/nix/store/vqhkmj60457j8nrw9vb94ln40rkic1p6-source`

### Notify maintainers

@Luflosi 

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/os-specific/linux/apfs/default.nix
+++ b/pkgs/os-specific/linux/apfs/default.nix
@@ -8,7 +8,7 @@
 }:
 
 let
-  tag = "0.3.12";
+  tag = "0.3.13";
 in
 stdenv.mkDerivation {
   pname = "apfs";
@@ -18,7 +18,7 @@ stdenv.mkDerivation {
     owner = "linux-apfs";
     repo = "linux-apfs-rw";
     rev = "v${tag}";
-    hash = "sha256-UuPWiqR1CiPlBmFleV+y3sowj82iUkgmakNM+T+Mlvc=";
+    hash = "sha256-PXTyPOZhBeFGXu71Jj46hlrgVemgmYrjHqkTFjTDoTc=";
   };
 
   hardeningDisable = [ "pic" ];

